src/OrchardCore.Modules/OrchardCore.ContentLocalization/Sitemaps/LocalizedContentItemsQueryProvider.cs

```diff
@@ -28,46 +28,46 @@ ILocalizationService localizationService
         _localizationService = localizationService;
     }
 
-    public async Task<ContentItemsQueryResult> GetContentItemsAsync(ContentTypesSitemapSource source, ContentItemsQueryContext context)
+    public async Task GetContentItemsAsync(ContentTypesSitemapSource source, ContentItemsQueryContext context, int? skip = null, int? take = null)
     {
         var routeableContentTypeDefinitions = await _routeableContentTypeCoordinator.ListRoutableTypeDefinitionsAsync();
         using var session = _store.CreateSession(withTracking: false);
 
+        IEnumerable<ContentItem> contentItems = null;
+
         if (source.IndexAll)
         {
             // Assumption here is that at least one content type will be localized.
             var ctdNames = routeableContentTypeDefinitions.Select(ctd => ctd.Name);
 
-            var results = await session.Query<ContentItem>()
-                .With<ContentItemIndex>(x => x.Published && x.ContentType.IsIn(ctdNames))
-                .OrderBy(x => x.CreatedUtc)
-                .ThenBy(x => x.Id)
-                .Skip(context.Skip)
-                .Take(context.Take)
-                .ListAsync();
-
-            return new ContentItemsQueryResult
+            if (take.HasValue && take > 0)
```

Comment:
You don't have to special-case. Just do `.Skip(skip ?? 0).Take(take ?? 0)` and it will do nothing more in the query if the values are `null`. Also `skip` or `take` without the other should be fine.

Reply:
in that case, we'll just make the parameter not nullable with a default value.

Reply:
done

Reply:
> in that case

Not "in that case", the arguments are what we want them to be. Does `0` have better meaning than `null` from the user's perspective? I'd say `null` might be easier to manage in general as a way to express "it's was not set". Maybe one day we'll want to know that `0` was passed explicitly. I'd rather keep `null` in this case, and use `?? 0` in Skip.

Reply:
I was just trying to avoid the null check every time. I'll make both parameters null again

Reply:
@sebastienros this was updated. Let me know if you see anything else otherwise please approve.